fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/GenericDataServiceImpl.java

```diff
@@ -262,4 +261,14 @@ private SqlRowSet getDatatableMetaData(final String datatable) {
             throw new DatatableNotFoundException(datatable);
         }
     }
+
+    private String formatDateTimeValue(String dateTimeValue) {
```

Comment:
Even though this solves the problem for now, this logic shouldn't be here.
On one note, we should definitely not decide based on the string representation of the datetime value, neither based on the length and whether it contains a `T`.

I looked at it a tiny bit deeper and we're completely missing the standardization on the datatable reading side.

If you look at `ReadWriteNonCoreDataServiceImpl#fillDatatableResultSetDataRows`, we're always reading the column value as string even though it's not necessarily a string.

In case of Date values, we could simply read them as Date type and then we don't have to deal with the non-sense DateTime parsing here.
It'd simplify a lot of things and set up proper boundaries on reading and processing the data.

Reply:
Yeah so my idea is to read dates as dates from the database and everything else as strings (for now at least) and use the existing date format on the API level so we don't break anything.

How I'd imagine, this is where we are right now:
Database -> string reading -> preparing the generic resultset -> format the strings -> send back on the API
Instead, let's do
Database -> string or date reading -> preparing the generic resultset -> format the date -> send back on the API

Hope that explains.